Block Editor: Update the Convert to Blocks dialog

Copons opened the pull request:
#### Changes proposed in this Pull Request

* Update the copy of the Convert to Blocks dialog.
* Update the modal to use the Core component.
* Add a new persistent `ui.convertToBlocksDialog` state that keeps track of dismissed Convert to Blocks dialogs, in order to only show it once per post.

| Before | After |
| - | - |
| <img width="617" alt="Image 2020-05-27 at 9 29 53 AM" src="https://user-images.githubusercontent.com/5634774/83024613-a52d3600-9ffc-11ea-881d-c2d182a5f4ff.png"> | <img width="735" alt="Screenshot 2020-06-02 at 11 25 02" src="https://user-images.githubusercontent.com/2070010/83510099-597b0080-a4c4-11ea-972f-97bfa852e442.png"> |

#### Testing instructions

* On a site using the Classic Editor, create a new post and write some content.
* Switch over to Gutenberg (Gutenframe).
* Make sure the "Convert to Blocks" notice shows up.
* Dismiss it (don't convert to blocks!) and reload the editor.
* Make sure the notice doesn't show up anymore.
* Switch back to Classic editor, and create another post with some content.
* Switch to Gutenberg, and make sure the notice shows up again.

Note: after switching to Gutenberg, adding **any** block content (including an empty Paragraph block) will automagically disable the notice as well.

Fixes https://github.com/Automattic/dotcom-manage/issues/87

a8ci18n commented on client/components/convert-to-blocks/index.jsx:
```diff
@@ -1,62 +1,68 @@
+			isVisible={ showDialog }
+			onClose={ onClose }
+		>
+			<h1>{ translate( 'Convert to blocks?' ) }</h1>
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'Convert to blocks' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/convert-to-blocks-dialog)

michelleweber commented:
Can we add some more info? I feel like we need to explain a bit more about the difference between "convert to blocks" and "add classic block" -- for someone not used to the block editor they both sound like "make some blocks."

Convert to blocks?
This post contains content you created with the classic editor. You can convert this content into blocks -- each paragraph and item will be its own block, that you can edit and move around -- or add a Classic block to simulate the classic editor.

scruffian commented on client/components/convert-to-blocks/index.jsx:
```diff
@@ -1,62 +1,68 @@
-		);
-	}
+	return (
+		<Dialog
```
What do you think about converting this to use `Modal` from Gutenberg?

Copons replied:
I've tried but it's missing lots of CSS. Maybe I'm missing something. 🤔

Copons commented:
Copy updated! Thanks @michelleweber!

<img width="616" alt="Screenshot 2020-06-01 at 19 09 57" src="https://user-images.githubusercontent.com/2070010/83439707-99e26c00-a43b-11ea-9a7f-16f57407648c.png">

michelleweber commented:
Thanks! Let's lose the comma after "block," it's been bugging me and it's not needed.

Also: would it be helpful to try and trim a little to get rid of the last single word on the last line, or does it vary with screen size so it doesn't matter so much?

Copons commented:
Updated the dialog to use the Core `Modal` component:

<img width="735" alt="Screenshot 2020-06-02 at 11 25 02" src="https://user-images.githubusercontent.com/2070010/83510099-597b0080-a4c4-11ea-972f-97bfa852e442.png">

Also @michelleweber I've removed the comma after "block".
And you guessed it right, the modal is responsive (both the previous Calypso version and this new Core one). 🙂

a8ci18n commented on client/components/convert-to-blocks/index.jsx:
```diff
@@ -1,62 +1,67 @@
+		<Modal
+			onRequestClose={ onClose( 'cancel' ) }
+			shouldCloseOnClickOutside={ false }
+			title={ translate( 'Convert to blocks?' ) }
```
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **2** times:
`translate( 'Convert to blocks' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/convert-to-blocks-dialog)

glendaviesnz commented on client/state/selectors/test/is-convert-to-blocks-dialog-dismissed.js:
```diff
@@ -0,0 +1,31 @@
+		expect( isDismissed ).toBe( false );
+	} );
+
+	test( 'should return false if the site and is not in state', () => {
```
should this be `should return false if the site is not in state` ?

glendaviesnz commented:
A rebase fixed the failing desktop test

Copons commented:
I've moved this back to WIP because the new `ui.convertToBlocksDialog` state should live in the upcoming modularised `editor` state introduced in #43388.
Once that PR is merged, I'll rebase this and move `convertToBlocksDialog` over to `editor`.

Copons commented:
In 38f8495 I've moved the new `convertToBlocksDialog` state from the `ui` root to the new modularized `editor`.

creativecoder commented:
@Copons Even though we're discussing reworking this prompt to "Convert to Blocks" (pb5gDS-DR-p2), I think we should go ahead with this change, as it's a nice improvement, and this potentially gets us ready for the deprecation sooner.

I added the "String Freeze" label to queue up the new strings for translation--I think we should wait for those before merging to prevent non-English users from seeing English text before it's translated.

creativecoder commented:
I've tested this change, and it works as expected:

- Clicking "Convert to Blocks" converts the content to blocks, modal does not appear on page reload
- Clicking "Stick with the Classic Block" or the X leaves content in the classic block, modal does not appear on page reload

One quirk is that in Firefox (Developer Edition), hovering over the X causes scrollbars to appear for the modal.

<img width="682" alt="Screen Shot 2020-07-01 at 17 50 14" src="https://user-images.githubusercontent.com/1699996/86298713-016b1300-bbc4-11ea-9ece-443f54249fc1.png">

Copons commented:
> One quirk is that in Firefox (Developer Edition), hovering over the X causes scrollbars to appear for the modal.

Oh, good catch, and it happens on Firefox "standard" and Chrome too.
Those scrollbars are caused by the Close button's tooltip being incorrectly positioned far below and to the right.

I believe it to be a Core bug.
We could fix by hiding the modal overflow, but I've chosen to remove the Close button instead.
It's not super useful anyway, and we really want folks to take a decision here, as opposed to just dismiss the modal (which makes the choice of using the Classic block in the user's stead)

glendaviesnz commented:
Decision has been made to remove this modal in favour of a an extra block toolbar option so closing for now.
